crates/relayer-types/src/core/ics24_host/identifier.rs

```diff
@@ -292,6 +292,50 @@ impl PartialEq<str> for ConnectionId {
     }
 }
 
+// A Vec<ConnectionId> wrapper for the multi-hop channels PoC implementation
+#[derive(Clone, Debug, Default, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize)]
+pub struct ConnectionIds(Vec<ConnectionId>);
+
+impl ConnectionIds {
+    pub fn as_slice(&self) -> &[ConnectionId] {
+        &self.0
+    }
+
+    pub fn into_vec(self) -> Vec<ConnectionId> {
+        self.0
+    }
+}
+
+impl FromStr for ConnectionIds {
+    type Err = ValidationError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let connection_ids = s
+            .split('/')
+            .map(|conn_id| conn_id.parse())
+            .collect::<Result<Vec<ConnectionId>, _>>()?;
+
+        if connection_ids.is_empty() {
+            return Err(ValidationError::empty());
+        }
+
+        Ok(Self(connection_ids))
+    }
+}
+
+impl Display for ConnectionIds {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), FmtError> {
+        let s = self
+            .0
+            .iter()
+            .map(|connection_id| connection_id.as_str())
+            .collect::<Vec<_>>()
+            .join(",");
+
+        write!(f, "{}", s)
```

Comment:
We can avoid allocating a new `Vec` before joining the strings by using the [`itertools` crate](https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.join)

```suggestion
        use itertools::Itertools;

        write!("{}", self.0.iter().join(","))
```